pull_request_guidelines.md

```diff
@@ -0,0 +1,78 @@
+## Why do we do pull requests?
+
+* Catching and reducing technical debt
+* Catching mistakes
+* Improving development skills
+* Familiarity with the whole codebase
+* Maintaining consistency
+
+## Pull Request Software
+
+* Reviewboard vs. Github
+
+## Creating Pull Requests
+
+* Pull request granularity (how big should my pull requests be?)
+ * Try to keep them under 500 lines.
```

Comment:
@weberbry I think I addressed this below.

@ryanmeador Maybe that first part is better suited to the `Git Workflow` section of sherpa. The second part could fit here, although it might be addressed below when we say to always rebase before merging.